File: acmicpc/acmicpc_21866.cs
// 21866 커피를 통해 추첨을 받자

public class Program
{
    public static void Main(string[] args)
    {
        List<int> list = (from item in Console.ReadLine().Split() select Convert.ToInt32(item)).ToList<int>();
        int[] scores = new int[] { 100, 100, 200, 200, 300, 300, 400, 400, 500 };


        for (int i = 0; i < list.Count; i++)
        {
            if (list[i] > scores[i])
            {
                Console.WriteLine("hacker"); 
                return;
            }
        }

        if (list.Sum() >= 100)
        {
            Console.WriteLine("draw");
        }
        else
        {
            Console.WriteLine("none");
        }
    }
}

File: Python/acmicpc-py/acmicpc_21866.py
# 21866 추첨을 통해 커피를 받자

def check(arr, coffee):
    for i in range(len(arr)):
        if arr[i] > coffee[i]:
            return "hacker"
        
    if sum(arr) < 100:
       return "none"

    return "draw"

arr = list(map(int, input().split()))
coffee = [100, 100, 200, 200, 300, 300, 400, 400, 500]
print(check(arr, coffee))

